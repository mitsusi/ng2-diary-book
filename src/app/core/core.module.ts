import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';

import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { AuthInterceptor } from './modules/http-interceptors/auth-append-token';

import {
  MatInputModule,
  MatButtonModule,
  MatIconModule,
  MatDatepickerModule,
  MatNativeDateModule,
  MatCardModule,
  MatSidenavModule,
  MatListModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
} from '@angular/material';

import { AppComponent } from './containers/app';
import { NotFoundPageComponent } from './containers/not-found-page';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { NotLoggedInComponent } from './components/not-logged-in';
import { CommonShowIfLoggedInComponent } from './containers/common-show-if-logged-in';
import { LoaderComponent } from './containers/loader';
import { BreadcrumbsComponent } from './components/breadcrumbs';
import { BreadcrumbsContainer } from './containers/breadcrumbs';
import { UserInfoContainer } from './containers/user-info';

import { BreadcrumbsService } from './services/breadcrumbs';

import { PipesModule } from '../shared/pipes';
import { EffectsModule } from '@ngrx/effects';
import { RouterEffects } from './effects/router';
import { FormsModule } from '@angular/forms';
import { TagsAutoCompleteModule } from './features/components/tags-autocomplete/tags-autocomplete.module';
import { TagsViewerModule } from './features/components/tags-viewer/tags-viewer.module';
import { RouterHelperService } from './services/router';
import { UserInfoComponent } from './components/user-info';
import { AuthService } from './services/auth.service';

export const COMPONENTS = [
  AppComponent,
  NotFoundPageComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
  NotLoggedInComponent,
  CommonShowIfLoggedInComponent,
  LoaderComponent,
  BreadcrumbsComponent,
  BreadcrumbsContainer,
  UserInfoContainer,
  UserInfoComponent,
];

const MAT_COMPONENTS = [
  MatInputModule,
  MatButtonModule,
  MatIconModule,
  MatDatepickerModule,
  MatNativeDateModule,
  MatCardModule,
  MatSidenavModule,
  MatListModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
];

@NgModule({
  imports: [
    CommonModule,
    FormsModule,
    RouterModule,
    PipesModule,
    HttpClientModule,
    EffectsModule.forRoot([RouterEffects]),
    TagsAutoCompleteModule,
    TagsViewerModule,
    [...MAT_COMPONENTS],
  ],
  declarations: COMPONENTS,
  exports: [
    ...COMPONENTS,
    CommonModule,
    FormsModule,
    HttpClientModule,
    TagsAutoCompleteModule,
    TagsViewerModule,
    [...MAT_COMPONENTS],
  ],
  providers: [
    RouterHelperService,
    BreadcrumbsService,
    {
      provide: HTTP_INTERCEPTORS,
      useClass: AuthInterceptor,
      multi: true,
    },
    AuthService,
  ],
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
    };
  }
}
