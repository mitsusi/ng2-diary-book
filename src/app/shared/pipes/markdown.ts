import { Pipe, PipeTransform } from '@angular/core';

import * as markdown from 'markdown-js';

@Pipe({ name: 'diaryMarkdown' })
export class MarkdownPipe implements PipeTransform {
  transform(textAsMarkdown: string = '') {
    return markdown.makeHtml(textAsMarkdown);
  }
}
