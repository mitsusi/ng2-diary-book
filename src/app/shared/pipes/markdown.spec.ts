import { MarkdownPipe } from './markdown';
import * as markdown from 'markdown-js';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Component, DebugElement, Input } from '@angular/core';
import { By } from '@angular/platform-browser';

describe('Markdown', () => {
  describe('as a class', () => {
    let pipe, makeHTMLSpy;

    beforeEach(() => {
      pipe = new MarkdownPipe();

      makeHTMLSpy = jest.spyOn(markdown, 'makeHtml');
    });
    afterEach(() => {
      makeHTMLSpy.mockRestore();
    });

    describe('test', () => {
      let result;

      beforeEach(() => {
        result = pipe.transform('kek');
      });

      it('should transform the input', () => {
        expect(result).toBe('<p>kek</p>');
      });

      it('should call the markdown library method', () => {
        expect(makeHTMLSpy).toBeCalledWith('kek');
      });
    });
  });

  describe('with a test component', function() {
    @Component({
      template: '<div>{{ text | diaryMarkdown }}</div>',
    })
    class TestComponent {
      @Input() text: string;
    }

    let component: TestComponent,
      fixture: ComponentFixture<TestComponent>,
      debugElement: DebugElement;

    beforeEach(() => {
      TestBed.configureTestingModule({
        declarations: [TestComponent, MarkdownPipe],
      });
    });

    beforeEach(() => {
      fixture = TestBed.createComponent(TestComponent);
      component = fixture.componentInstance;
      debugElement = fixture.debugElement.query(By.css('div'));

      fixture.detectChanges();
    });

    it('snapshot is correct', () => {
      component.text = `
        # Header
        ## Second level header

        - bob
        - kek
        - lul

        How are you?
        ===============

        Another header
        ---------------------------

        **bob**
      `;

      fixture.detectChanges();

      expect(fixture).toMatchSnapshot();
    });

    it('snapshot is correct 22228', () => {
      component.text = 'header h1\n===\n\nabc\n\n*italic* **bold**';

      fixture.detectChanges();

      expect(fixture).toMatchSnapshot();
    });

    it('span contains correct stuff', () => {
      component.text = '*italic* **bold**';

      fixture.detectChanges();

      expect(debugElement.nativeElement.textContent).toBe(
        '<p><em>italic</em> <strong>bold</strong></p>',
      );
    });
  });
});
