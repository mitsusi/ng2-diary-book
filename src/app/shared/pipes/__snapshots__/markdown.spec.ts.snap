// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown with a test component snapshot is correct 1`] = `
<ng-component
  text={[Function String]}
>
  <div>
    <pre><code>    # Header ## Second level header - bob - kek - lul How are you? =============== Another header --------------------------- **bob** </code></pre>
  </div>
</ng-component>
`;

exports[`Markdown with a test component snapshot is correct 22228 1`] = `
<ng-component
  text={[Function String]}
>
  <div>
    <h1>header h1</h1> <p>abc</p> <p><em>italic</em> <strong>bold</strong></p>
  </div>
</ng-component>
`;
